templates/includes/pagination.html

```diff
@@ -1,17 +1,34 @@
 {% if is_paginated %}
   <ul class="pagination">
-    {% if page_obj.has_previous %}
-      <li class="page-item">
-        <a href="?page= {{ page_obj.previous_page_number }}" class="page-link">prev</a>
-      </li>
-    {% endif %}
+    <li class="page-item">
+      {% if page_obj.has_previous %}
+        <a class="page-link" href="?page={{ page_obj.previous_page_number }}">Prev</a>
+      {% else %}
+        <span class="page-link disabled-link">Prev</span>
+      {% endif %}
+    </li>
     <li class="page-item active">
-      <span class="page-link">{{ page_obj.number }} of {{ paginator.num_pages }}</span>
+      <span class="page-link">
+        <input id="page-number" type="text" value="{{ page_obj.number }}"> of {{ paginator.num_pages }}</span>
+    </li>
+    <li class="page-item">
+      {% if page_obj.has_next %}
+        <a class="page-link" href="?page={{ page_obj.next_page_number }}">Next</a>
+      {% else %}
+        <span class="page-link disabled-link">Next</span>
+      {% endif %}
     </li>
-    {% if page_obj.has_next %}
-      <li class="page-item">
-        <a href="?page= {{ page_obj.next_page_number }}" class="page-link">next</a>
-      </li>
-    {% endif %}
   </ul>
+  <script>
+    let page_input = document.getElementById("page-number")
+    page_input.onchange = function () { page_input.size = page_input.value.length || 1; }
+    page_input.onchange();
+    page_input.addEventListener("keyup", event => {
+      if (event.key !== "Enter") {
+        page_input.onchange();
+        return;
+      }
+      window.location.href = `?page=${page_input.value}`
+    });
+  </script>
```

Comment:
what is the purpose of this script?
isn't it enough to use html and css to do the same with if statements??

Reply:
This is to dynamically resize the input when text is entered.